src/puppetlabs/puppetdb/scf/partitioning.clj

```diff
@@ -65,106 +60,31 @@
         date-formatter (DateTimeFormatter/ISO_OFFSET_DATE_TIME)
 
         table-name-suffix (date-suffix date)
-        full-table-name (format "%s_%s" base-table table-name-suffix)]
-    (apply jdbc/do-commands
-           (concat [(format (str "CREATE TABLE IF NOT EXISTS %s ("
-                                 (str/join ", "
-                                          (cons "CHECK ( %s >= TIMESTAMP WITH TIME ZONE '%s' AND %s < TIMESTAMP WITH TIME ZONE '%s' )"
-                                                (constraint-fn table-name-suffix)))
-                                 ") INHERITS (%s)")
-                            full-table-name
-                            ;; this will write the constraint in UTC. note: when you read this back from the database,
-                            ;; you will get it in local time.
-                            ;; example: constraint will have 2019-09-21T00:00:00Z but upon querying, you'll see 2019-09-21 17:00:00-07
-                            ;; this is just the database performing i18n
-                            date-column (.format start-of-day date-formatter) date-column (.format start-of-next-day date-formatter)
-                            base-table)]
-                   (index-fn full-table-name table-name-suffix)))))
+        full-table-name (format "%s_%s" base-table table-name-suffix)
+        {current-pg-version :version} (sutils/db-metadata)
+        allow-expressions sutils/declarative-partitions-allow-expressions
+        ;; XXX although, if this is acceptable in pg11, then presumably date-formatter is doing all the work and the inclusion of TIMESTAMP WITH TIME ZONE is unnecessary here?
+        range-template (if (neg? (compare current-pg-version allow-expressions))
```

Comment:
I suspect this either unnecessary or broken.

Reply:
Using 'foo'::timestamptz is not also not allowed in pg11. It has to be a literal constant. But since we are formatting them as ISO_OFFSET_DATE_TIME which has tz info ('2007-12-03T10:15:30+01:00') we shouldn't need the additional cast. I'll remove it and version test.